R/iso_save.R

```diff
@@ -0,0 +1,53 @@
+#' Save data as R Data Structure (.rds)
+#'
+#' This function saves the passed in iso_files to an R Data Structure (.rds) file, which is a fairly efficient compressed data storage format. Data exported this way can be easily read back into isoreader using the standard \code{\link{iso_read_continuous_flow}} and \code{\link{iso_read_dual_inlet}} functions.
+#'
+#' @inheritParams iso_get_raw_data
+#' @param filepath the path (folder and filename) to the export file. The correct file extension is automatically added if not already in the filename, i.e. filename can be provided with or without extension.
+#' @family export functions
+#' @return returns the iso_files object invisibly for use in pipelines
+#' @export
+iso_save <- function(iso_files, filepath, quiet = default(quiet)) {
+
+  # safety checks
+  if(!iso_is_object(iso_files)) stop("can only export iso files or lists of iso files", call. = FALSE)
+  filepath <- get_rds_export_filepath(iso_files, filepath)
+
+  # save iso_files
+  if (!quiet) {
+    sprintf("Info: exporting data from %d iso_files into R Data Archive '%s'",
+            length(iso_as_file_list(iso_files)),
+            str_replace(filepath, "^\\.(/|\\\\)", "")) %>% message()
+  }
+
+  saveRDS(iso_files, file = filepath)
+  return(invisible(iso_files))
+}
+
+#' Export data to R Data Archive (.rds)
+#'
+#' Deprecated in favour of `iso_save()'
+#'
+#' @inheritParams iso_get_raw_data
+#' @param filepath the path (folder and filename) to the export file. The correct file extension is automatically added if not already in the filename, i.e. filename can be provided with or without extension.
+#' @family export functions
+#' @return returns the iso_files object invisibly for use in pipelines
+#' @export
+#' @seealso iso_save
+iso_export_to_rds <- function(iso_files, filepath, quiet = default(quiet)) {
+  warning("`iso_export_to_rds' is deprecated and will call `iso_save()'. Please call `iso_save()' directly to avoid this warning.")
```

Comment:
```suggestion
  warning("`iso_export_to_rds' is deprecated and will call `iso_save()'. Please call `iso_save()' directly to avoid this warning.", call. = FALSE, immediate. = TRUE)
```